_posts/2023-06-19-End-to-end-acknowledgements-in-Data-Prepper.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title:  "End-to-end acknowledgements in Data Prepper"
+authors:
+- kkondaka
+- dlv
+date: 2023-06-19 10:00:00 -0700
+categories:
+  - technical-post
+
+meta_keywords: Data Prepper E2E, Data Prepper sink, Data Prepper pipeline as the sink
+meta_description: Learn how Data Prepper provides data durability through the use of end to end acknowledgements by delivering data to the sink before notifying the source.
+
+---
+
+[Data Prepper](https://opensearch.org/docs/latest/data-prepper/index/) is an open-source data collector used to ingest data into OpenSearch clusters. While Data Prepper has an in-memory buffer that allows for fast throughput, users also want improved data durability, particularly, confirmation that the data received by Data Prepper reaches the desired sink. To solve challenges with data durability, Data Prepper now has capability to use end-to-end acknowledgements.
+
+Data Prepper maintainers and their teams have observed that a common challenge to data durability is the reliability of the ingestion pipeline. For example, if the OpenSearch cluster cannot receive data because of temporary stress on the cluster or underscaling, then Data Prepper cannot send data to the destination sink. Further, Data Prepper may exhaust memory or other system resources or the hardware running Data Prepper may fail resulting the data loss during the ingestion.
```

Comment:
```suggestion
The Data Prepper maintainers and their teams have observed that a common challenge to data durability is the reliability of the ingestion pipeline. For example, if the OpenSearch cluster cannot receive data because of temporary stress on the cluster or underscaling, then Data Prepper cannot send data to the destination sink. Further, Data Prepper may exhaust memory or other system resources or the hardware running Data Prepper may fail resulting the data loss during the ingestion.
```